src/browser-provider/index.js

```diff
@@ -0,0 +1,53 @@
+import { assignIn } from 'lodash';
+import { PathBrowserProvider, LocalBrowserProvider, RemoteBrowserProvider } from './built-in';
+
+
+export var BUILT_IN_PROVIDERS = {
+    'local':  LocalBrowserProvider,
+    'path':   PathBrowserProvider,
+    'remote': RemoteBrowserProvider
+};
+
+
+export default class BrowserProvider {
+    constructor (providerName) {
+        var provider = BUILT_IN_PROVIDERS[providerName];
+
+        if (!provider)
+            provider = require(`testcafe-provider-${providerName}`);
```

Comment:
It's better move require logic out of here to `bootstrapper` and rename it to `BrowserProviderHost` like we have for reporters.